### ISSUE TITLE
Override JaversSqlRepository in JaversSqlAutoConfiguration

### ISSUE THREAD

**Issue by fcanovas1986** [javers/javers#1335]
**Is your feature request related to a problem? Please describe.**

I am trying to override the bean **JaversSqlRepository**, which has a `@ConditionalOnMissingBean`

https://github.com/javers/javers/blob/6c847776f8316ad88e1e3f8a6e4dab7efb2dc314/javers-spring-boot-starter-sql/src/main/java/org/javers/spring/boot/sql/JaversSqlAutoConfiguration.java#L77

I have created a custom repository which extends from **JaversSqlRepository**, in order to add some observability features and register some metrics.

```
@Repository
public class CustomJaversSqlRepository extends JaversSqlRepository {

  public CustomJaversSqlRepository(final SessionFactory sessionFactory,
      final CommitMetadataRepository commitRepository,
      final GlobalIdRepository globalIdRepository,
      final CdoSnapshotRepository cdoSnapshotRepository,
      final CdoSnapshotFinder finder, final JaversSchemaManager schemaManager,
      final SqlRepositoryConfiguration sqlRepositoryConfiguration) {
    super(sessionFactory, commitRepository, globalIdRepository, cdoSnapshotRepository, finder, schemaManager, sqlRepositoryConfiguration);
  }
}
```

The point is, that to create a **JaversSqlRepository** is used the **SqlRepositoryBuilder** which in last instance call a method `build()` that configures some important things (adding JaversModules and components) for the correct Javers working. I guess that just calling the public constructor there will be some things not configured.

https://github.com/javers/javers/blob/6c847776f8316ad88e1e3f8a6e4dab7efb2dc314/javers-spring-boot-starter-sql/src/main/java/org/javers/spring/boot/sql/JaversSqlAutoConfiguration.java#L89

https://github.com/javers/javers/blob/6c847776f8316ad88e1e3f8a6e4dab7efb2dc314/javers-persistence-sql/src/main/java/org/javers/repository/sql/SqlRepositoryBuilder.java#L114

To make this working, actually I had to create a new **JaversModule**.

```
public class CustomJaversModule implements JaversModule {
  private static final Class[] moduleComponents = new Class[]{
      CustomJaversSqlRepository.class
  };

  @Override
  public Collection<Class> getComponents() {
    return Arrays.asList(moduleComponents);
  }
}
```
And then create a new **CustomSqlRepositoryBuilder** to make it work with my **CustomJaversSqlRepository**.

```
public class CustomJaversSqlRepositoryBuilder extends SqlRepositoryBuilder {

  public static CustomJaversSqlRepositoryBuilder sqlRepository() {
    return new CustomJaversSqlRepositoryBuilder();
  }

  @Override
  protected <T> T getContainerComponent(final Class<T> ofClass) {

    if (ofClass.equals(JaversSqlRepository.class)) {

      this.addModule(new CustomJaversModule());
      return (T) super.getContainerComponent(CustomJaversSqlRepository.class);
    }

    return super.getContainerComponent(ofClass);
  }

}
```

With this, finally I can create my bean in my configuration:

```
  @Bean(name = "JaversSqlRepositoryFromStarter")
  public JaversSqlRepository customSqlRepository(final ConnectionProvider connectionProvider,
      final JaversSqlProperties javersSqlProperties, final DialectName javersSqlDialectName) {

    return CustomJaversSqlRepositoryBuilder
        .sqlRepository()
        .withSchema(javersSqlProperties.getSqlSchema())
        .withConnectionProvider(connectionProvider)
        .withDialect(javersSqlDialectName)
        .withSchemaManagementEnabled(javersSqlProperties.isSqlSchemaManagementEnabled())
        .withGlobalIdCacheDisabled(javersSqlProperties.isSqlGlobalIdCacheDisabled())
        .withGlobalIdTableName(javersSqlProperties.getSqlGlobalIdTableName())
        .withCommitTableName(javersSqlProperties.getSqlCommitTableName())
        .withSnapshotTableName(javersSqlProperties.getSqlSnapshotTableName())
        .withCommitPropertyTableName(javersSqlProperties.getSqlCommitPropertyTableName())
        .build();
  }
```

In summary, my question is if there is any other alternative I am not aware of to override the bean **JaversSqlRepository**?? Because the  method `build()` of **SqlRepositoryBuilder** must be call to set correctly the repository. Is there any other easier way to get advantage of that `@ConditionalOnMissingBean`  rather than override all the configuration mentioned above. Am i missing something? or this is the best I can do with that `@ConditionalOnMissingBean` ?

Does it make sense that `@ConditionalOnMissingBean` if that bean cannot be really overriden?

**Comment by bartoszwalacik** [comment]
Hi @fcanovas1986 ,
for sure you can override the `JaversSqlRepository` bean, as long as your bean extends the JaversSqlRepository class, it's a standard Spring feature. What is unclear for me is why you try to use SqlRepositoryBuilder, as this class is not designed to build anyting other then JaversSqlRepository.

The intention of this `@ConditionalOnMissingBean` is provide an option to build the JaversSqlRepository with custom configuration, rather than injecting a custom impl. You can still inject a custom impl but then you are on your own.
